### PR TITLE
 KEYMAPPER: Actions can now be bound to joystick buttons 

### PR DESCRIPTION
**Pull request by bgK**
Here finally comes the juicy bit:
Capture d’écran de 2020-01-26 19-23-56

This is the last major part of the keymapper related changes I had planned. There definitely are some remaining issues, and new low hanging fruits to be plucked, but I feel like those can be dealt with directly in tree.

### CODE REVIEW

**Comment by bluegr** [comment]
After merging the changes in PR #2012, this can no longer be merged cleanly. Could you please fix the conflicts?

**Comment by bgK** [comment]
I'll merge this tomorrow unless someone needs more time to review.

File: backends/events/sdl/sdl-events.cpp [hunk 1]
```diff
@@ -983,91 +903,27 @@ int SdlEventSource::mapSDLJoystickButtonToOSystem(Uint8 sdlButton) {
...
-	if (ev.jbutton.button == JOY_BUT_LMOUSE) {
-		event.type = Common::EVENT_LBUTTONDOWN;
-		return processMouseEvent(event, _km.x / MULTIPLIER, _km.y / MULTIPLIER);
-	} else if (ev.jbutton.button == JOY_BUT_RMOUSE) {
-		event.type = Common::EVENT_RBUTTONDOWN;
-		return processMouseEvent(event, _km.x / MULTIPLIER, _km.y / MULTIPLIER);
-	} else {
-		event.type = Common::EVENT_KEYDOWN;
-		switch (ev.jbutton.button) {
-		case JOY_BUT_ESCAPE:
```
Comment (ccawley2011) [inline backends/events/sdl/sdl-events.cpp]: Since these defines are no longer used, they can be removed from the top of the file.

**Comment by bgK** [comment]
Merged manually as df7ce0c55f6802093b2d11d46dc96e34d1a0ee05...376a77e735acb1001e75ed0f6dc87de086842a16

**Comment by rsn8887** [comment]
@bgK 

Can you please confirm that the official default mappings here:
https://wiki.scummvm.org/index.php?title=PlayStation_Vita
and here:
https://wiki.scummvm.org/index.php?title=Nintendo_Switch
are still intact?

It seems to me like you removed the default joystick mapping of at least  "space", "dot", "escape", as well as the diagonal keypad buttons, and maybe more.

Am I right that if I map a game where it says "default map" then it saves the mapping for all games using the default map?

Where can I map the diagonals (KP9 etc.)?

**Pull request closed without merging**